MapboxNavigation/NavigationMapView.swift

```diff
@@ -266,6 +276,20 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         addGestureRecognizer(mapTapGesture)
         
         installUserCourseView()
+        
+        styleObservation = observe(\.style, options: .new) { [weak self] (mapView, change) in
+            guard change.newValue != nil else { return }
+            guard let self = self else { return }
+            
+            if let routes = self.routes, let currentRoute = routes.first {
+                self.show(routes)
+                self.showWaypoints(on: currentRoute)
+            }
```

Comment:
It would be better to gradually reduce the usage of `NavigationMapView.routes`. We’ve successfully removed NavigationMapView’s listening for route progress change. In the same vein, we should aim to remove extra state that the map view keeps around that could easily get out of sync with RouteController.